pkg/sidecar/sidecar.go

```diff
@@ -183,6 +232,131 @@ func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	return nil
 }
 
+func (s *Sidecar) writeJSON(fileName string, certs map[string]interface{}) {
+	file, err := json.Marshal(certs)
+	if err != nil {
+		s.config.Log.Errorf("Unable to parse certs: %v", err)
+	}
+
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	if err = os.WriteFile(jsonPath, file, os.ModePerm); err != nil {
+		s.config.Log.Errorf("Unable to write JSON file: %v", err)
+	}
+}
+
+func (s *Sidecar) updateJWTBundle(jwkSet *jwtbundle.Set) {
+	s.config.Log.Debug("Updating JWT bundle")
+
+	bundles := make(map[string]interface{})
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			s.config.Log.Errorf("Unable to marshal JWT bundle: %v", err)
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	s.writeJSON(s.config.JWTBundleFilename, bundles)
+	s.config.Log.Info("JWT bundle updated")
+}
+
+func (s *Sidecar) fetchJWTSVID(options ...workloadapi.ClientOption) (*jwtsvid.SVID, error) {
+	clientOptions := workloadapi.WithClientOptions(options...)
+
+	jwtSource, err := workloadapi.NewJWTSource(context.Background(), clientOptions)
+	if err != nil {
+		s.config.Log.Errorf("Unable to create JWTSource: %v", err)
+		return nil, err
+	}
+	defer jwtSource.Close()
+
+	jwtSVID, err := jwtSource.FetchJWTSVID(context.Background(), jwtsvid.Params{Audience: s.config.JWTAudience})
```

Comment:
context must be provided by caller

Reply:
@JU4N98 This comment has not been addressed yet.

Reply:
Hey @faisal-memon  i don't know why github isn't showing the change, but the branch of my repo has the change. Here is the [link](https://github.com/JU4N98/spiffe-helper/blob/d6bd491a6be671521832e3c1559ecb1628ca8402/pkg/sidecar/sidecar.go#L269-L292) to that part. Same for the previous comment.